### PR TITLE
Add Apple Silicon support to CI

### PR DESCRIPTION
**Pull request by jpfeuffer**
Todo:
Make distinction for GitHub between arm and x8664 in:
- [x] packaging, to upload packages for both and name them different (naming should prob. already done by cmake correctly)
-  [x] ccache or other caches or anything that depends on the OS name only but should depend on OS+arch. Maybe we can use synonymous macos-14-arm64 runner label or so?
- [ ] pyopenms and maybe conda (still on Jenkins anyway)

### CODE REVIEW

**Comment by jpfeuffer** [comment]
We should really switch to qt6 soon. It's getting harder and harder to find support for new platforms for 5.

**Comment by jpfeuffer** [comment]
Can someone finish up smaller hiccups if there are any?

**Comment by poshul** [comment]
> Can someone finish up smaller hiccups if there are any?

Yeah, I can try to get this over the finish line.

File: .github/workflows/openms_ci_matrix_full.yml [hunk 1]
```diff
@@ -187,18 +201,16 @@ jobs:
         fi
 
         if [[ "${{ matrix.os }}" == macos-* ]]; then
-          # Qt5_DIR set by Install Qt5 step
-          echo "cmake_prefix=$Qt5_DIR/lib/cmake;$Qt5_DIR" >> $GITHUB_OUTPUT
           brew install --quiet ccache autoconf automake libtool ninja && brew link --overwrite ccache
-          brew install libsvm xerces-c boost eigen sqlite coinutils cbc cgl clp
+          brew install libsvm xerces-c boost eigen sqlite coinutils cbc cgl clp qt@5
+          echo "cmake_prefix=$(brew --prefix qt@5)/lib/cmake;$(brew --prefix qt@5)" >> $GITHUB_OUTPUT
+          echo "Qt5_DIR=$(ls $(brew --prefix qt@5)/*)/lib/cmake/Qt5" >> $GITHUB_ENV
```
Comment (poshul) [inline .github/workflows/openms_ci_matrix_full.yml]: Is this supposed to go to GITHUB_OUTPUT or GITHUB_ENV? ENV is currently choking on it.

File: .github/workflows/openms_ci_matrix_full.yml [hunk 2]
```diff
@@ -138,8 +152,8 @@ jobs:
         # NOTE: x64 is hardcoded. No support for 32bit
         arch: x64
 
-    - if: startsWith(matrix.os, 'macos') || startsWith(matrix.os, 'windows')
-      name: Install Qt (Windows and macOS)
+    - if: startsWith(matrix.os, 'windows')
```
Comment (poshul) [inline .github/workflows/openms_ci_matrix_full.yml:155]: Why are we moving away from this action for Mac? I'm unclear how this relates to adding apple silicon.

**Review by poshul — CHANGES_REQUESTED** [review]
Can we separate out moving away from the Install-qt-action into another PR? I'm not sure why it's in here, and it currently breaks the build.

File: .github/workflows/openms_ci_matrix_full.yml (hunk 1 above)
Comment (poshul) [reply .github/workflows/openms_ci_matrix_full.yml]: Hmm, so *that's* what happens if there is a push while I'm writing a comment.

**Comment by poshul** [comment]
> We should really switch to qt6 soon. It's getting harder and harder to find support for new platforms for 5.

@jpfeuffer @timosachsenberg how much developer time do we think that's going to take? Is it possible to parcel out into sprint tasks?

File: .github/workflows/openms_ci_matrix_full.yml (hunk 2 above)
Comment (jpfeuffer) [reply .github/workflows/openms_ci_matrix_full.yml:155]: see commit message. qt5 not available.

**Comment by jpfeuffer** [comment]
Hard to say. See #6252 
Sprint tasks can be the steps in the porting guide.
Then tasks for each affected module.

**Comment by jpfeuffer** [comment]
I think this is ready to merge and can be fixed together with your current efforts. I think there is not much left, if at all.

**Comment by poshul** [comment]
Do we need to add anything for the KNIME update site specifically for ARM?

**Review by poshul — APPROVED** [review]

**Review by timosachsenberg — APPROVED** [review]
awesome! thanks

**Comment by jpfeuffer** [comment]
Yes a little. See #7355 and the comment in the description on GKN.

**Pull request merged**